plugin/storage/integration/integration.go

```diff
@@ -299,6 +299,7 @@ func (s *StorageIntegration) testFindTraces(t *testing.T) {
 		}
 		expectedTracesPerTestCase = append(expectedTracesPerTestCase, expected)
 	}
+	time.Sleep(time.Second)
```

Comment:
@yurishkuro I attempted to address the [test failure](https://github.com/jaegertracing/jaeger/actions/runs/8739681859/job/23982877531?pr=5349) occurring with ES7 by incorporating t.Parallel() to ensure the test keeps retrying until 100 seconds have elapsed. Despite this effort, the test continued to fail. While I understand that relying on time.Sleep is not ideal, I'm currently exploring alternative solutions.

Reply:
Sleep alone is not a reliable mechanism to deal with concurrency, retries are. We already have sleep inside `waitForCondition`, which is used by `findTracesByQuery` called in this test. Why doesn't that work?

Reply:
It turns out that the retry condition in waitForCondition was incorrect. The retry condition should match not only the trace count but also the span count. I addressed this issue in commit 8c8f2e3495b396d41a5191aae1ee3af9ad4671d6. I'm curious why the test only fails intermittently when using other Elasticsearch versions.